react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      index: 0
+    };
+    this.arrowClicks = this.arrowClicks.bind(this);
+    this.renderProgressDots = this.renderProgressDots.bind(this);
+    this.clickDots = this.clickDots.bind(this);
+    this.updateCarousel = this.updateCarousel.bind(this);
+  }
+
+  componentDidMount() {
+    this.updateCarousel();
+  }
+
+  updateCarousel() {
+    const intervalId = setInterval(() => {
+      if (this.state.index === this.props.images.length - 1) {
+        this.setState({
+          index: 0
+        });
+      } else {
+        this.setState({
+          index: this.state.index + 1
+        });
+      }
+    }, 3000);
+    return intervalId;
+  }
+
+  arrowClicks(event) {
+    clearInterval(this.updateCarousel());
```

Comment:
Here, instead of calling `updateCarousel()` right away, first clear `this.interval`.